src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContextFactory.cs

```diff
@@ -23,6 +27,6 @@ internal interface IWorkspaceProjectContextFactory
         /// <param name="projectGuid">Project guid.</param>
         /// <param name="hierarchy">Obsolete. The argument is ignored.</param>
         /// <param name="binOutputPath">Initial project binary output path.</param>
-        IWorkspaceProjectContext CreateProjectContext(string languageName, string projectUniqueName, string projectFilePath, Guid projectGuid, object hierarchy, string binOutputPath);
+        Task<IWorkspaceProjectContext> CreateProjectContextAsync(string languageName, string projectUniqueName, string projectFilePath, Guid projectGuid, object hierarchy, string binOutputPath);
```

Comment:
@jasonmalinowski  should we take a CancellationToken here?  Seems... odd  i think this is a very stateful change.  I'm not sure what it would mean to be cancellable.

Reply:
but maybe we shoudl take it just in case?

Reply:
I'd add it only if the project system team says they need/want it. This shim only exists for their needs.